src/order/atoms.lean

```diff
@@ -97,6 +118,121 @@ lemma is_coatom_iff_is_atom_dual : is_coatom a ↔ is_atom (order_dual.to_dual a
 
 end atoms
 
+section atomic
+
+variables (α) [bounded_lattice α]
+
+/-- A lattice is atomic iff every element other than `⊥` has an atom below it. -/
+class is_atomic : Prop :=
+(eq_bot_or_exists_atom_le : ∀ (b : α), b = ⊥ ∨ ∃ (a : α), is_atom a ∧ a ≤ b)
+
+/-- A lattice is coatomic iff every element other than `⊤` has a coatom above it. -/
+class is_coatomic : Prop :=
+(eq_top_or_exists_le_coatom : ∀ (b : α), b = ⊤ ∨ ∃ (a : α), is_coatom a ∧ b ≤ a)
+
+export is_atomic (eq_bot_or_exists_atom_le) is_coatomic (eq_top_or_exists_le_coatom)
+
+variable {α}
+
+theorem is_atomic_iff_is_coatomic_dual : is_atomic α ↔ is_coatomic (order_dual α) :=
```

Comment:
Should this and the following be reversed and `simp`? (I'm not familiar enough with the `order` library to know if `order_dual` is something we usually try to `simp` away.)

Reply:
I think not usually, but it seems like a good thing to try.